### ISSUE TITLE
Format check fails but `nix fmt` brings no changes.

### ISSUE THREAD

**Issue by zmrocze** [numtide/treefmt-nix#156]
**Describe the bug**

Hi! 

Format check fails. But running the formatter doesn't modify anything. It's all formatted but check fails.

**To Reproduce**

At [this commit](https://github.com/mlabs-haskell/spo-anywhere/tree/e4e17961406a9d4d54e80c9b29439b4fcacca618) first run:

```bash
nix fmt
```

Observe `0 files changed` output.

Then run:

```
nix build .\#checks.x86_64-linux.treefmt
```

The build fails.

**Expected behavior**

The build doesn't fail or the formatter formats.

**System information**

[This flake](https://github.com/mlabs-haskell/spo-anywhere/tree/e4e17961406a9d4d54e80c9b29439b4fcacca618). That is:

- `treefmt 0.6.0`
- `nix 2.18.0`
- `treefmt-nix` rev `8cd95da6c30852adb2a06c4b6bdacfe8b64a0`, used with nixpkgs:
```
    treefmt-nix = {
      url = "github:numtide/treefmt-nix";
      inputs.nixpkgs.follows = "nixpkgs";
    };
```

**Comment by montchr** [comment]
I am also encountering this issue with `treefmt` v2.0.1, running via [cachix/git-hooks.nix](https://github.com/cachix/git-hooks.nix).

**Comment by montchr** [comment]
https://github.com/numtide/treefmt/issues/342

**Comment by yangm97** [comment]
@montchr I don't think that issue is related but I do have a reproducer on tsandrini/flake-parts-builder#62 which looks more like OP's report.
